src/bdecode.cpp

```diff
@@ -479,7 +479,7 @@ namespace libtorrent
 		{
 			bdecode_token const& t = tokens[token];
 			TORRENT_ASSERT(t.type == bdecode_token::string);
-			int const size = m_root_tokens[token + 1].offset - t.offset - t.start_offset();
```

Comment:
did you try replacing ``m_root_tokens`` by ``tokens``? Does that not generate the efficient code?